### PR TITLE
[ioctl] Refactor nodedelegate command in new ioctl

### LINKED ISSUE #3279: [ioctl] refactor nodedelegate command in new ioctl
### What would you like to be added
 - Step 1: refactor  command  [newcmd/node/nodedelegate.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/node/nodedelegate.go) 
 - Step 2: append necessary unit test case in [newcmd/node/nodedelegate_test.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/node/nodedelegate_test.go) 

#### Notes:
1. Use [function of client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) . 
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/node/nodedelegate.go ).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `nodedelegate` command in new `ioctl`, with the following note.

* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3279 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] TestNewNodeDelegateCmd

**Test Configuration**:

- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Newman1515 — CHANGES_REQUESTED** [review]
Ithink

**Comment by LuckyPigeon** [comment]
> Ithink

Yes?

**Review by CoderZhi — APPROVED** [review]

File: ioctl/newcmd/node/nodedelegate.go [hunk 1]
```diff
@@ -100,57 +109,57 @@ func NewNodeDelegateCmd(c ioctl.Client) *cobra.Command {
...
 				if err != nil {
 					sta, ok := status.FromError(err)
 					if ok {
-						return output.NewError(output.APIError, sta.Message(), nil)
+						return errors.New(sta.Message())
 					}
-					return output.NewError(output.NetworkError, "failed to invoke ReadState api", err)
+					return errors.Wrap(err, "failed to invoke ReadState api")
 				}
 				var BPs state.CandidateList
```
Comment (huof6829) [inline ioctl/newcmd/node/nodedelegate.go:160]: lowercase for local variables

File: ioctl/newcmd/node/nodedelegate.go [hunk 2]
```diff
@@ -167,20 +176,20 @@ func NewNodeDelegateCmd(c ioctl.Client) *cobra.Command {
...
 					if err != nil {
-						return output.NewError(0, "failed to get chain meta", err)
+						return errors.Wrap(err, "failed to get chain meta")
 					}
-					_epochNum = chainMeta.Epoch.Num
+					epochNum = chainMeta.Epoch.Num
 				}
 
-				response, err := bc.GetEpochMeta(_epochNum, c)
+				response, err := bc.GetEpochMeta(epochNum, c)
```
Comment (huof6829) [inline ioctl/newcmd/node/nodedelegate.go:190]: delete space line

File: ioctl/newcmd/node/nodedelegate.go [hunk 3]
```diff
@@ -88,7 +96,8 @@ func NewNodeDelegateCmd(c ioctl.Client) *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			cmd.SilenceUsage = true
 			var err error
```
Comment (huof6829) [inline ioctl/newcmd/node/nodedelegate.go:98]: `err` is not used seemly.

File: ioctl/newcmd/node/nodedelegate.go [hunk 4]
```diff
@@ -100,57 +109,57 @@ func NewNodeDelegateCmd(c ioctl.Client) *cobra.Command {
...
 				if err != nil {
-					return output.NewError(0, "failed to get chain meta", err)
+					return errors.Wrap(err, "failed to get chain meta")
 				}
-				_epochNum = chainMeta.Epoch.Num + 1
-				message := nextDelegatesMessage{Epoch: int(_epochNum)}
+				epochNum = chainMeta.Epoch.Num + 1
+				message := nextDelegatesMessage{Epoch: int(epochNum)}
 
 				ctx := context.Background()
```
[the remark this answers is not in the document]
Comment (huof6829) [reply ioctl/newcmd/node/nodedelegate.go:117]: also add in  bc.GetProbationList  and  bc.GetEpochMeta

File: ioctl/newcmd/node/nodedelegate.go [hunk 5]
```diff
@@ -189,20 +198,20 @@ func NewNodeDelegateCmd(c ioctl.Client) *cobra.Command {
 					StartBlock:  int(epochData.Height),
 					TotalBlocks: int(response.TotalBlocks),
 				}
-				probationListRes, err := bc.GetProbationList(_epochNum, c)
+				probationListRes, err := bc.GetProbationList(epochNum, c)
```
Comment (huof6829) [inline ioctl/newcmd/node/nodedelegate.go:201]: bc.GetProbationList(client, epochNum)  should be clear

File: ioctl/newcmd/node/nodedelegate.go [hunk 6]
```diff
@@ -167,20 +176,20 @@ func NewNodeDelegateCmd(c ioctl.Client) *cobra.Command {
...
 					if err != nil {
-						return output.NewError(0, "failed to get chain meta", err)
+						return errors.Wrap(err, "failed to get chain meta")
 					}
-					_epochNum = chainMeta.Epoch.Num
+					epochNum = chainMeta.Epoch.Num
 				}
 
-				response, err := bc.GetEpochMeta(_epochNum, c)
+				response, err := bc.GetEpochMeta(epochNum, c)
```
Comment (huof6829) [inline ioctl/newcmd/node/nodedelegate.go:189]: bc.GetEpochMeta(client, epochNum)  should be clear

File: ioctl/newcmd/node/nodedelegate.go (hunk 3 above)
Comment (LuckyPigeon) [reply ioctl/newcmd/node/nodedelegate.go:98]: used in here
https://github.com/iotexproject/iotex-core/blob/56c8c09f05e58ad4b2469f39912eccab3bd368da/ioctl/newcmd/node/nodedelegate.go#L234-L236

**Review by huof6829 — APPROVED** [review]

**Pull request merged**